### DISCUSSION TITLE
Obtener el valor de un optional en un findFirst

### DISCUSSION THREAD

**Discussion by juanmpereira** [dds-jv/foro#132] [Consultas de objetos] [answered]
Al querer filtrar para devover una prenda particular que acepte el clima y ""combine"" con las otras seleccionadas escribo esto:
' return SonDeCategoria(cat).filter(p->p.esAptaParaElClima(temperatura) && p.combina(prendaSuperior)).findFirst();'

y me tira error diciendo que lo que devuelve es Optional<Prenda> y no Prenda. Como puedo solucionar esto?

**Comment by gastonprieto** [comment]
Buenas!

Optional es un tipo de Java, que mensajes entiende?

**Comment by juanmpereira** [comment]
Claro, el tema es que yo eso que devuelvo lo asigno a un objeto Prenda. Entonces me da error si devuelve Optional pero que necesito que devuelva Prenda. Como puedo hacer para solucionarlo?

**Comment by gastonprieto** [comment]
Jeje, no me respondiste la pregunta :P

Que mensajes entiende el objeto Optional? eso debería ayudarte para conseguir lo que necesitas.

**Comment by juanmpereira** [comment]
Ahi estuve viendo un poco mas de este objeto en la pagina de oracle. Dice que este objeto entiende el null pointer. Con lo cual deberia poner a las prendas como Optional<Prenda> y para mostrarlo usar un prenda.isPresent()? A esto de null te referias con que mensaje entiende?

**Comment by gastonprieto** [comment]
> Dice que este objeto entiende el null pointer

No entendí a que te referis con esto, null pointer es una excepción.

> para mostrarlo usar un prenda.isPresent()

isPresent() es un mensaje que sirva para decirte si el optional tiene un valor o no. No entendí que tiene que ver con mostrarlo.

> A esto de null te referias con que mensaje entiende?

No, mensajes son las operaciones que expone un objeto para que lo puedan utilizar, me refería a cuales son los que tiene, por ejemplo recié nombraste uno que es #isPresent, aunque no parece ser el que necesitas.

----
Optional es un tipo que representa a un objeto que puede o no tener valor, es como una caja, que puede o no tener algo adentro.

**Comment by juanmpereira** [comment, marked as answer]
Ah perfecto, ahi estuve viendo. El que necesitaria usar es el get que me devuelve objeto en caso de existir y una excepcion en caso contrario?

**Comment by gastonprieto** [comment]
Exacto!

**Comment by juanmpereira** [comment]
gracias!
